whois-api/src/main/java/net/ripe/db/whois/api/httpserver/RemoteAddressCustomizer.java

```diff
@@ -4,43 +4,104 @@
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jetty.http.BadMessageException;
 import org.eclipse.jetty.server.Connector;
 import org.eclipse.jetty.server.HttpConfiguration;
 import org.eclipse.jetty.server.Request;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 import javax.annotation.Nullable;
 import java.net.InetSocketAddress;
+import java.util.Arrays;
 import java.util.Enumeration;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 /**
- * When HTTP requests are proxied via a loadbalancer, the X-Forwarded-For value will replace the request address.
- * In case of multiple (comma separated) values in X-Forwarded-For, the last value is used.
+ * When HTTP requests via trusted source use clientIp query paramater as remote Address if set
  */
-public class RemoteAddressCustomizer  implements HttpConfiguration.Customizer {
-
-    private static final Logger LOGGER = LoggerFactory.getLogger(RemoteAddressCustomizer.class);
+public class RemoteAddressCustomizer implements HttpConfiguration.Customizer {
 
     private static final Splitter COMMA_SPLITTER = Splitter.on(',').omitEmptyStrings().trimResults();
+    public static final String QUERY_PARAM_CLIENT_IP = "clientIp";
+    private final Set<Interval> trusted;
+
+    // if client address is set in X-Forwarded-For header by HTTP proxy
+    private final boolean usingForwardedForHeader;
+
+    public RemoteAddressCustomizer(final String trustedIpRanges, final boolean xForwardedFor) {
+        this.trusted = getIntervals(trustedIpRanges.split(","));
+        this.usingForwardedForHeader = xForwardedFor;
+    }
 
     @Override
     public void customize(final Connector connector, final HttpConfiguration httpConfiguration, final Request request) {
-        request.setRemoteAddr(InetSocketAddress.createUnresolved(getRemoteAddress(request), request.getRemotePort()));
+        String remoteAddress = formaliseAddress(getRemoteAddrFromRequest(request));
+
+        if (isTrusted(remoteAddress)){
+            String clientIp = getClientIp(request);
+            if (clientIp != null){
+                remoteAddress = clientIp;
+            }
+        }
+        request.setRemoteAddr(InetSocketAddress.createUnresolved(remoteAddress, request.getRemotePort()));
     }
 
     @Nullable
-    private String getRemoteAddress(final Request request) {
-        final Enumeration<String> headers = request.getHeaders(HttpHeaders.X_FORWARDED_FOR);
-        if (headers == null || !headers.hasMoreElements()) {
-            return request.getRemoteAddr();
+    private String getClientIp(final Request request){
+        try {
+            final String clientIp = request.getParameter(QUERY_PARAM_CLIENT_IP);
+            if (StringUtils.isNotEmpty(clientIp)) {
+                return clientIp;
+            }
+        } catch (BadMessageException ex){
+            //ignore
         }
+        return null;
+    }
+
+    private String formaliseAddress(final String address){
```

Comment:
Rename to `normaliseAddress` or `stripSquareBrackets`

Reply:
I will but why not formalise???

Reply:
Done

Reply:
"formalise" isn't really the correct term, normalise is better (i.e. convert to a normal form) or be explicit about just removing square brackets. Thanks